src/wp-includes/comment.php

```diff
@@ -408,32 +393,25 @@ function get_comment_count( $post_id = 0 ) {
 		'all'                 => 0,
 	);
 
-	foreach ( $totals as $row ) {
-		switch ( $row['comment_approved'] ) {
-			case 'trash':
-				$comment_count['trash'] = $row['total'];
-				break;
-			case 'post-trashed':
-				$comment_count['post-trashed'] = $row['total'];
-				break;
-			case 'spam':
-				$comment_count['spam']            = $row['total'];
-				$comment_count['total_comments'] += $row['total'];
-				break;
-			case '1':
-				$comment_count['approved']        = $row['total'];
-				$comment_count['total_comments'] += $row['total'];
-				$comment_count['all']            += $row['total'];
-				break;
-			case '0':
-				$comment_count['awaiting_moderation'] = $row['total'];
-				$comment_count['total_comments']     += $row['total'];
-				$comment_count['all']                += $row['total'];
-				break;
-			default:
-				break;
-		}
+	$where = '';
+	if ( $post_id > 0 ) {
+		$where = $wpdb->prepare( ' AND comment_post_ID = %d', $post_id );
 	}
+	$comment_count = array(
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		'approved'            => $wpdb->get_var( "SELECT COUNT( * ) FROM {$wpdb->comments} WHERE comment_approved = '1'{$where}" ),
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		'awaiting_moderation' => $wpdb->get_var( "SELECT COUNT( * ) FROM {$wpdb->comments} WHERE comment_approved = '0'{$where}" ),
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		'spam'                => $wpdb->get_var( "SELECT COUNT( * ) FROM {$wpdb->comments} WHERE comment_approved = 'spam'{$where}" ),
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		'trash'               => $wpdb->get_var( "SELECT COUNT( * ) FROM {$wpdb->comments} WHERE comment_approved = 'trash'{$where}" ),
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		'post-trashed'        => $wpdb->get_var( "SELECT COUNT( * ) FROM {$wpdb->comments} WHERE comment_approved = 'post-trashed'{$where}" ),
+	);
```

Comment:
```suggestion
		$args = array(
			'count'                     => true,
			'update_comment_meta_cache' => false,
		);
		if ( $post_id > 0 ) {
			$args['post_id'] = $post_id;
		}
		$mapping       = array(
			'approved'            => 'approve',
			'awaiting_moderation' => 'hold',
			'spam'                => 'spam',
			'trash'               => 'trash',
			'post-trashed'        => 'post-trashed',
		);
		$comment_count = array();
		foreach ( $mapping as $key => $value ) {
			$comment_count[ $key ] = get_comments( array_merge( $args, array( 'status' => $value ) ) );
		}
```

Instead of using raw queries here, we should just use get_comments / WP_Comment_Query. This has filters and caching for performance and makes this easier to maintain.

Reply:
Added your suggestion.